libglusterfs/src/hashfn.c

```diff
@@ -137,7 +137,7 @@ gf_dm_hashfn(const char *msg, int len)
 
     for (i = 0; i < full_quads; i++) {
         for (j = 0; j < 4; j++) {
-            word = *intmsg;
+        	word = le32toh(*intmsg);
```

Comment:
'word' is an 'int', le32toh gives us an unsigned int (32bit), 'array' is uint32_t as well. Perhaps 'word' should be uint32_t as well?

Reply:
done